costmap_2d/CMakeLists.txt

```diff
@@ -103,7 +102,7 @@ add_library(costmap_2d
   src/footprint.cpp
   src/costmap_layer.cpp
 )
-add_dependencies(costmap_2d ${catkin_EXPORTED_TARGETS})
+add_dependencies(costmap_2d costmap_2d_generate_messages_cpp ${catkin_EXPORTED_TARGETS})
```

Comment:
Is generate_messages_cpp not one of hte exported targets?

Reply:
catkin exports are generated targets from dependencies, generated messages generates all generated c++ from the current target.